sbi/inference/base.py

```diff
@@ -45,19 +45,18 @@ def __init__(
                 ['slice', 'hmc', 'nuts'].
         """
 
-        self._warn_on_possibly_batched_observations(true_observation)
-        
         # XXX want self._true_observation (atleast_2d) as attribute instead?
-        self._simulator = set_simulator_attributes(simulator, prior,
-                                                   true_observation)
-        self._true_observation = atleast_2d(true_observation)
+        self._simulator, self._prior, self._true_observation = prepare_sbi_problem(
+            simulator, 
+            prior, 
+            # XXX: move atleast2d to the user input checks? 
+            # without it most of our functional tests fail because of missing 
+            # batch dim in data.
+            atleast_2d(true_observation)
```

Comment:
The problem is that we agreed on basically rejecting / throwing an error if `observed_data` is not batched. We could relax this requirement in that we correct for an unbatched observation using `.unsqueeze(0)` or so?

Reply:
We really have to talk about this 1:1 to prevent misunderstandings. As far as I have understood so far:
- we don't reject if observed_data is not batched but has dimension 1 - we wrap.
- we can do `at_least2d` explicitly in the function we are calling (where you can also comment and see it in the context of other transformations/checks, test more easily, etc). Doing it in the signature feels really strange. What is the problem with that?

Reply:
OK agreed, we will correct for unbatched observed data using `atleast_2d` in the `proces_observed_data()` function.